keyboards/boston_meetup/2019/2019.c

```diff
@@ -0,0 +1,226 @@
+    draw_string(MOD_DISPLAY_X + 3, MOD_DISPLAY_Y + 2, "S", PIXEL_OFF, NORM, 0);
+  } else {
+    draw_string(MOD_DISPLAY_X + 3, MOD_DISPLAY_Y + 2, "S", PIXEL_ON, NORM, 0);
+  }
+  if (mods & MOD_LCTL) {
+    draw_rect_filled_soft(MOD_DISPLAY_X + 10, MOD_DISPLAY_Y, 5 + (1 * 6), 11, PIXEL_ON, NORM);
+    draw_string(MOD_DISPLAY_X + 13, MOD_DISPLAY_Y + 2, "C", PIXEL_OFF, NORM, 0);
+  } else {
+    draw_string(MOD_DISPLAY_X + 13, MOD_DISPLAY_Y + 2, "C", PIXEL_ON, NORM, 0);
+  }
+  if (mods & MOD_LALT) {
+    draw_rect_filled_soft(MOD_DISPLAY_X + 20, MOD_DISPLAY_Y, 5 + (1 * 6), 11, PIXEL_ON, NORM);
+    draw_string(MOD_DISPLAY_X + 23, MOD_DISPLAY_Y + 2, "A", PIXEL_OFF, NORM, 0);
+  } else {
+    draw_string(MOD_DISPLAY_X + 23, MOD_DISPLAY_Y + 2, "A", PIXEL_ON, NORM, 0);
+  }
+  if (mods & MOD_LGUI) {
+    draw_rect_filled_soft(MOD_DISPLAY_X + 30, MOD_DISPLAY_Y, 5 + (1 * 6), 11, PIXEL_ON, NORM);
+    draw_string(MOD_DISPLAY_X + 33, MOD_DISPLAY_Y + 2, "G", PIXEL_OFF, NORM, 0);
+  } else {
+    draw_string(MOD_DISPLAY_X + 33, MOD_DISPLAY_Y + 2, "G", PIXEL_ON, NORM, 0);
+  }
+
+/* Lock display is 23 x 32 */
+#define LOCK_DISPLAY_X 104
+#define LOCK_DISPLAY_Y 0
+
+  if (led_numlock == true) {
+    draw_rect_filled_soft(LOCK_DISPLAY_X, LOCK_DISPLAY_Y, 5 + (3 * 6), 9, PIXEL_ON, NORM);
+    draw_string(LOCK_DISPLAY_X + 3, LOCK_DISPLAY_Y + 1, "NUM", PIXEL_OFF, NORM, 0);
+  } else if (led_numlock == false) {
+    draw_string(LOCK_DISPLAY_X + 3, LOCK_DISPLAY_Y + 1, "NUM", PIXEL_ON, NORM, 0);
+  }
+  if (led_capslock == true) {
+    draw_rect_filled_soft(LOCK_DISPLAY_X + 0, LOCK_DISPLAY_Y + 11, 5 + (3 * 6), 9, PIXEL_ON, NORM);
+    draw_string(LOCK_DISPLAY_X + 3, LOCK_DISPLAY_Y + 11 +1, "CAP", PIXEL_OFF, NORM, 0);
+  } else if (led_capslock == false) {
+    draw_string(LOCK_DISPLAY_X + 3, LOCK_DISPLAY_Y + 11 +1, "CAP", PIXEL_ON, NORM, 0);
+  }
+
+  if (led_scrolllock == true) {
+    draw_rect_filled_soft(LOCK_DISPLAY_X + 0, LOCK_DISPLAY_Y + 22, 5 + (3 * 6), 9, PIXEL_ON, NORM);
+    draw_string(LOCK_DISPLAY_X + 3, LOCK_DISPLAY_Y + 22 +1, "SCR", PIXEL_OFF, NORM, 0);
+  } else if (led_scrolllock == false) {
+    draw_string(LOCK_DISPLAY_X + 3, LOCK_DISPLAY_Y + 22 +1, "SCR", PIXEL_ON, NORM, 0);
+  }
+  send_buffer();
+}
+
+void read_host_led_state(void) {
+  uint8_t leds = host_keyboard_leds();
+  if (leds & (1 << USB_LED_NUM_LOCK))    {
+    if (led_numlock == false){
+    led_numlock = true;}
+    } else {
+    if (led_numlock == true){
+    led_numlock = false;}
+    }
+  if (leds & (1 << USB_LED_CAPS_LOCK))   {
+    if (led_capslock == false){
+    led_capslock = true;}
+    } else {
+    if (led_capslock == true){  
+    led_capslock = false;}
+    }
+  if (leds & (1 << USB_LED_SCROLL_LOCK)) {
+    if (led_scrolllock == false){
+    led_scrolllock = true;}
+    } else {
+    if (led_scrolllock == true){
+    led_scrolllock = false;}
+    }
+}
+
+uint32_t layer_state_set_kb(uint32_t state) {
+  state = layer_state_set_user(state);
+  layer = biton32(state);
```

Comment:
This doesn't look like it's used? 

Also, not sure you need this, since `layer_state` is a global variable that you can pull from anywhere.

Reply:
its so if there is a change in layer (timed event or what not) the OLED would receive the change state and send for a new refersh to the display